### PR TITLE
Enterprise Integration documentation enhancements

### PR DESCRIPTION
**Pull request by calebhailey**
- [x] Clean up Sensu Enterprise integration documentation 
- [x] Document the Sensu Enterprise integration client definition attributes
- [x] New "configuration scope" explanations 
- [x] More examples for nested configuration scopes

### CODE REVIEW

**Comment by calebhailey** [comment]
Getting started adding the Sensu Enterprise integration attributes to the client definition specification. This is going to be a big help for our Enterprise customers (and our support team). 😄

**Comment by calebhailey** [comment]
This PR also includes some better explanations around configuration scopes, including new examples for nested configuration scopes...

**Comment by calebhailey** [comment]
This PR is ready for final review/merge...

File: source/docs/0.23/clients.md [hunk 1]
```diff
@@ -409,12 +417,17 @@ redact
     ~~~
 : example
   : ~~~ shell
-    "redact": ["password", "ec2_access_key", "ec2_secret_key"]
+    "redact": [
+      "password",
+      "ec2_access_key",
+      "ec2_secret_key"
+    ]
```
Comment (agoddard) [inline source/docs/0.23/clients.md:424]: 👍

**Comment by agoddard** [comment]
:shipit:

**Comment by calebhailey** [comment]
:boom:

**Pull request merged**